File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorDIAntipattern/_2_ServiceLocator/Services/MsSqlBasedRepository.cs
using ServiceLocatorDIAntipattern._2_ServiceLocator.Interfaces;
using Microsoft.Practices.Unity;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Services
{
  public interface IRepository
  {
    void Save(AcmeMessage message);
  }

  public class MsSqlBasedRepository : IRepository
  {
    private final DataDestination _sqlDataDestination = ApplicationRoot.Context.Resolve<SqlDataDestination>();

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}

File: Java/dependency-injection/_04_bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Services/MsSqlBasedRepository.java
package com.github.grzesiek_galezowski.BastardInjection.Services;

import com.github.grzesiek_galezowski.BastardInjection.Interfaces.Message;
import com.github.grzesiek_galezowski.BastardInjection.Interfaces.DataDestination;

public class MsSqlBasedRepository implements IRepository {
  private final DataDestination _dataDestination;

  public MsSqlBasedRepository() {
    this(new SqlDataDestination());
  }

  //for tests
  public MsSqlBasedRepository(DataDestination dataDestination) {
    _dataDestination = dataDestination;
  }

  public void save(Message message) {
    message.writeTo(_dataDestination);
  }
}

